builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,270 @@
+	name := d.Get("name").(string)
+	request := &rds.DescribeEventSubscriptionsInput{
+		SubscriptionName: aws.String(name),
+	}
+
+	describeResp, err := rdsconn.DescribeEventSubscriptions(request)
+	if err != nil {
+		if rdserr, ok := err.(awserr.Error); ok && rdserr.Code() == "SubscriptionNotFound" {
+			log.Printf("[WARN] No RDS Event Subscription by name (%s) found", name)
+			d.SetId("")
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error reading RDS Event Subscription %s: %s", name, err)
+	}
+
+	if len(describeResp.EventSubscriptionsList) != 1 {
+		return nil, fmt.Errorf("Unable to find RDS Event Subscription: %#v", describeResp.EventSubscriptionsList)
+	}
+
+	return describeResp.EventSubscriptionsList[0], nil
+}
+
+func resourceAwsDbEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+
+	// check/update tags, missing from go-aws-sdk?
+	if d.HasChange("enabled") || d.HasChange("event_categories") ||
+		d.HasChange("sns_topic") || d.HasChange("source_type") {
+		eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+		eventCategories := make([]*string, eventCategoriesSet.Len())
+		for i, eventCategory := range eventCategoriesSet.List() {
+			eventCategories[i] = aws.String(eventCategory.(string))
+		}
+
+		// Do we want to wait while it is in "modifying" state?
+
+		_, err := rdsconn.ModifyEventSubscription(&rds.ModifyEventSubscriptionInput{
+			SubscriptionName: aws.String(d.Id()),
+			Enabled:          aws.Bool(d.Get("enabled").(bool)),
+			SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+			SourceType:       aws.String(d.Get("source_type").(string)),
+			EventCategories:  eventCategories,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	deleteOpts := rds.DeleteEventSubscriptionInput{
+		SubscriptionName: aws.String(d.Id()),
+	}
+
+	if _, err := rdsconn.DeleteEventSubscription(&deleteOpts); err != nil {
+		rdserr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+
+		if rdserr.Code() != "DBEventSubscriptionNotFoundFault" {
+			return err
+		}
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func resourceAwsDbEventSubscriptionRefreshFunc(
+	d *schema.ResourceData,
+	meta interface{}) resource.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		sub, err := resourceAwsDbEventSubscriptionRetrieve(d, meta)
+
+		if err != nil {
+			log.Printf("Error on retrieving DB Event Subscription when waiting: %s", err)
+			return nil, "", err
+		}
+
+		if sub == nil {
```

Comment:
The current behaviour is probably reasonable here, I was more asking about the motivation.

We have [`NotFoundChecks` in `resource.StateChangeConf`](https://github.com/hashicorp/terraform/blob/master/helper/resource/state.go#L57-L60) set to `20` by default - i.e. if this happens more than 20 times, an error is returned, otherwise this function is recalled.